lib/active_record/precounter.rb

```diff
@@ -27,7 +27,7 @@ def precount(*association_names)
         end
 
         count_by_id = if reflection.has_scope?
-                        reflection.scope_for(reflection.klass.where(reflection.inverse_of.name => records.map(&:id))).group(
+                        reflection.scope_for(reflection.klass).where(reflection.inverse_of.name => records.map(&:id)).group(
```

Comment:
Actually `#scoped_for` would accept both `klass` and `relation` as long as responding to `#unscoped`.

The most important difference between older and newer versions is that older version would unscope the passed relation.
This change has an effect to keep the condition for older version, so we can add failing test case.

At least this change has one undesireable effect for newer version.

```ruby
      def scope_for(relation, owner = nil)
        relation.instance_exec(owner, &scope) || relation
      end
```

The `#scope_for` expects that the receiver of `instance_exec` is a relation, so this change may break existing correct code, so preferably pass a relation like `reflection.klass.unscoped` unlike `klass`.

Reply:
Thanks to comment. Then I unscoped that https://github.com/k0kubun/activerecord-precounter/commit/daf6f498af3792c3d45fe4c83c88d6ee11748baf and confirmed at least it's passing the current specs with activerecord 5.0 https://travis-ci.org/k0kubun/activerecord-precounter/jobs/451721874.